### DISCUSSION TITLE
List of event via eme_events empty since PHP upgrade

### DISCUSSION THREAD

**Discussion by nmueller78** [liedekef/events-made-easy#269] [General]
Last week I had the hoster upgrade PHP on my web server from 7.4 to 8.2.8. Since then, [eme_events] no longer returns any results. Specifically, I use [eme_events scope=this_year] and [eme_events scope=next_year].
Other shortcodes I use seem to work without problems.

Unfortunately, I can't get to the log files directly on the server, so I can't provide concrete error messages easily.

Version information:
Wordpress 6.3
Events Made Easy 2.4.5

**Comment by liedekef** [comment]
Both placeholders+options work fine here ... Best is indeed to get to the logfiles somehow.

**Comment by nmueller78** [comment]
OK, it took a few, but I now have a concrete error message:

Fatal error: Uncaught TypeError: Unsupported operand types: string + int in /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-content/plugins/events-made-easy/eme-events.php:4240
Stack trace:
 #0 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-content/plugins/events-made-easy/eme-events.php(4650): eme_get_events_list()
 #1 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-includes/shortcodes.php(395): eme_get_events_list_shortcode()
 #2 [internal function]: do_shortcode_tag()
 #3 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-includes/shortcodes.php(235): preg_replace_callback()
 #4 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-includes/class-wp-hook.php(310): do_shortcode()
 #5 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-includes/plugin.php(205): WP_Hook->apply_filters()
 #6 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-includes/post-template.php(256): apply_filters()
 #7 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-content/themes/olivewp/template-parts/content-page.php(27): the_content()
 #8 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-includes/template.php(787): require('/usr/local/http...')
 #9 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-includes/template.php(720): load_template()
 #10 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-includes/general-template.php(206): locate_template()
 #11 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-content/themes/olivewp/page.php(103): get_template_part()
 #12 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-includes/template-loader.php(106): include('/usr/local/http...')
 #13 /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-blog-header.php(19): require_once('/usr/local/http...')
 #14 /usr/local/httpd/www.bdmp-mv.de/htdocs/index.php(17): require('/usr/local/http...')
 #15 {main} thrown in /usr/local/httpd/www.bdmp-mv.de/htdocs/wp-content/plugins/events-made-easy/eme-events.php on line 4240

Version information:
Events Made Easy: 2.4.8
Wordpress: 6.3.1

**Comment by liedekef** [comment]
Hum ... that seems a new php strict thing ...
This should fix it: https://github.com/liedekef/events-made-easy/commit/45e3a1fea3de226a9a46efa7cc5f9840117a86de 
==> the change is bigger (because I corrected more), but in your case just the added line 4113 (`$limit = intval($limit);`) should be sufficient.

**Reply by liedekef** [reply]
I also found the real problem: your EME option "Number of events to show in lists" is probably empty (should be a number) ==> I'll correct that in the next version.

**Reply by nmueller78** [reply]
Yes, the EME option "Number of events to show in lists" was indeed empty. I entered a 0 and the problem was solved. The list is displayed again.
Thanks a lot
